ui/app/rif/components/table/pagination.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+const defaultButton = props => <button {...props}>{props.children}</button>;
+
+export default class Pagination extends Component {
+  constructor (props) {
+    super(props);
+    this.changePage = this.changePage.bind(this);
+    this.state = {
+      visiblePages: this.getVisiblePages(null, props.pages),
+    };
+  }
+
+  static propTypes = {
+    pages: PropTypes.number,
+    page: PropTypes.number,
+    PageButtonComponent: PropTypes.any,
+    onPageChange: PropTypes.func,
+    previousPage: PropTypes.func,
+    nextPage: PropTypes.func,
+    canNextPage: PropTypes.bool,
+    canPreviousPage: PropTypes.bool,
+    className: PropTypes.any,
+  };
+
+  componentWillReceiveProps (nextProps) {
+    this.changePage(nextProps.page + 1);
+  }
+
+  filterPages = (visiblePages, totalPages) => {
+    return visiblePages.filter(page => page <= totalPages);
+  };
+
+  changePage (page) {
+    const activePage = this.props.page + 1;
+
+    if (page === activePage) {
+      return;
+    }
+
+    const visiblePages = this.getVisiblePages(page, this.props.pages);
+
+    this.setState({
+      visiblePages: this.filterPages(visiblePages, this.props.pages),
+    });
+
+    this.props.onPageChange(page - 1);
+  }
+
+  getVisiblePages = (page, total) => {
```

Comment:
Why is this needed? isnt this handled by react table by default?

Reply:
You need to send a pagination component for the react table, you can use it to change the pages. All the examples in react-table doc was using something like this. You can set the page size, and other things, but there's no way to get the visible pages

Reply:
I don't know about this, seems to me that maybe those were examples because react table should resolve this for you instead of you making the pagination by hand, it looks weird

Reply:
Yesterday i talked to @marcosmartinez7, this is for the custom pagination, React-Table only manages functions like goToPage, and an index for the pages, if you want to code something like we need in the Figma design (Page's indexes shown), we need to use this pagination component.